### PR TITLE
refactor: publisher queue

### PR DESCRIPTION
**Pull request by ganchoradkov**
# Description
- Reworked the publisher `queue` to remove items on relay ACK instead of when the request is responded to
- Removed unused deps from a few packages

## How Has This Been Tested?
Integration tests
dogfooding
canary `2.7.4-rc-a7ad32`

## Due Diligence

- [ ] Breaking change
- [ ] Requires a documentation update
  - [ ] Related docs issue/PR (if docs are not included in this PR):
- [ ] Requires a e2e/integration test update

### CODE REVIEW

File: packages/core/src/constants/relayer.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ export const RELAYER_CONTEXT = "relayer";
 
 export const RELAYER_EVENTS = {
   message: "relayer_message",
+  message_ack: "message_ack",
```
Comment (bkrem) [inline packages/core/src/constants/relayer.ts]: Nit: this value should be `relayer_` prefixed for the sake of consistency with the other values in the mapping.

File: packages/core/test/publisher.spec.ts [hunk 2]
```diff
@@ -23,24 +28,130 @@ describe("Publisher", () => {
...
+      // Emit ACKs
+      items.forEach((item) =>
+        core.relayer.events.emit(RELAYER_EVENTS.message_ack, { id: item.opts.id }),
+      );
+
+      // -> Queue should clear after the ACKs.
+      expect(publisher.queue.size).to.equal(0);
+
+      // emit multiple pulses to ensure queue is cleared
+      await throttle(100);
```
Comment (bkrem) [inline packages/core/test/publisher.spec.ts]: Why throttle before the first heartbeat pulse here rather than after?

**Review by bkrem — APPROVED** [review]
Nice, great to see the publisher/queue getting some love 💯 

This feels like a definite candidate for wider pre-release canary testing (e.g. in web3modal lab). Do you think it makes sense to release the smaller patches that were merged earlier separately and then do a release for this?

File: packages/core/test/publisher.spec.ts (hunk 2 above)
Comment (ganchoradkov) [reply packages/core/test/publisher.spec.ts]: left over from different iterations of the test. Its redundant and I removed it 💯

**Pull request merged**